web/src/main/java/org/apache/shiro/web/servlet/SimpleCookie.java

```diff
@@ -362,8 +362,14 @@ public String readValue(HttpServletRequest request, HttpServletResponse ignored)
         String value = null;
         javax.servlet.http.Cookie cookie = getCookie(request, name);
         if (cookie != null) {
-            value = cookie.getValue();
-            log.debug("Found '{}' cookie value [{}]", name, value);
+            // Validate that the cookie is used at the correct place.
+            String path = StringUtils.clean(getPath());
+            if (path != null && !request.getRequestURI().startsWith(path)) {
```

Comment:
I might be missing something, but if `getPath()` returns a non-null/non-empty string, then `calculatePath(request)` would return that value, which would trivially be equal?